### PR TITLE
Support nullable types when reading Avro files [HZ-2507]

### PR DESCRIPTION
**Pull request by burakgok**
Specifically, the following statements are affected:
```sql
CREATE MAPPING <name> TYPE File OPTIONS ('format'='avro', ...)
TABLE(AVRO_FILE(...))
```

Fixes HZ-2507

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases

### CODE REVIEW

**Comment by k-jamroz** [comment]
Is it a breaking change?

**Review by k-jamroz — APPROVED** [review]

**Comment by burakgok** [comment]
This change just enables Avro file mappings and `AVRO_FILE` function to read unions of a primitive and null. It doesn't break the existing capability of reading primitives.

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/file/SqlAvroTest.java [hunk 1]
```diff
@@ -187,6 +184,15 @@ public void test_schemaDiscovery() {
 
     @Test
     public void test_tableFunction() {
+        test_tableFunction(AVRO_FILE);
+    }
+
+    @Test
+    public void test_tableFunction_nullableFields() {
+        test_tableFunction(AVRO_NULLABLE_FILE);
+    }
```
Comment (frant-hartm) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/file/SqlAvroTest.java:193]: We are missing test reading `null`s. This reads nullable fields, which contain values.

**Review by frant-hartm — CHANGES_REQUESTED** [review]

**Comment by burakgok** [comment]
run-lab-run

**Comment by burakgok** [comment]
run-lab-run

**Comment by burakgok** [comment]
run-lab-run

**Review by frant-hartm — APPROVED** [review]

**Pull request merged**